numpy/linalg/linalg.py

```diff
@@ -2832,3 +2830,98 @@ def _multi_dot(arrays, order, i, j, out=None):
         return dot(_multi_dot(arrays, order, i, order[i, j]),
                    _multi_dot(arrays, order, order[i, j] + 1, j),
                    out=out)
+
+
+# diagonal
+
+def _diagonal_dispatcher(x, /, *, offset=None):
+    return (x,)
+
+
+@array_function_dispatch(_diagonal_dispatcher)
+def diagonal(x, /, *, offset=0):
+    """
+    Returns specified diagonals of a matrix (or a stack of matrices) ``x``.
+
+    This function is Array API compatible, contrary to
+    :py:func:`numpy.diagonal`.
```

Comment:
Seeing this only because these new additions to the numpy API triggered astropy warnings, and perhaps too late since it is already merged, but wouldn't it be better to tell what the difference is here rather than what is here now? As is, this comment forces the user to either look up the array API or to look at the docstring of `np.diagonal` to find out that it is the axes that are used. Maybe add something like
"""
This function is Array API compatible, contrary to
:py:func:`numpy.diagonal`: the matrix is assumed
to be defined by the last two dimensions.
"""

Reply:
You're right - it's clearer about the difference. I will update the docstring.

Reply:
Thanks!